File: src/main/java/io/github/cheatbook/amzrevenuemanager/domain/entity/SkuName.java
package io.github.cheatbook.amzrevenuemanager.domain.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * SKU名エンティティです。
 */
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SkuName {

    /**
     * SKU
     */
    @Id
    private String sku;

    /**
     * 日本語名
     */
    private String japaneseName;

    /**
     * 親SKU
     */
    private String parentSku;
}

File: backend-typescript-lambda/src/domain/models/SkuName.ts
/**
 * SKU名マッピングインターフェース
 */
export interface SkuName {
  /** SKU */
  sku: string;
  /** 日本語名 */
  japaneseName: string;
  /** 親SKU */
  parentSku?: string;
  /** 更新日時 */
  updatedAt?: string;
}
